src/anndata/_settings.py

```diff
@@ -25,13 +28,46 @@ class DeprecatedOption(NamedTuple):
     removal_version: str | None
 
 
-# TODO: inherit from Generic[T] as well after python 3.9 is no longer supported
-class RegisteredOption(NamedTuple):
-    option: str
-    default_value: T
-    doc: str
-    validate: Callable[[T], bool] | None
-    type: object
+def describe(self: RegisteredOption, *, rst: bool = False) -> str:
+    if rst:
+        default_str = repr(self.default_value).replace("\\", "\\\\")
+        doc = f"""\
+        .. attribute:: settings.{self.option}
+           :type: {getattr(self.type, "__name__", None) or str(self.type)}
+           :value: {default_str}
+
+           {self.description}
+        """
+    else:
+        type_str = self.type.__name__ if isinstance(self.type, type) else str(self.type)
+        doc = f"""\
+        {self.option}: `{type_str}`
+            {self.description} (default: `{self.default_value!r}`).
+        """
+    return textwrap.dedent(doc)
+
+
+if sys.version_info >= (3, 10):
```

Comment:
this is better than a comment, since Ruff’s pyupgrade (UP) rule will autoremove the other branch once we drop 3.9 support.